src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java

```diff
@@ -254,14 +257,37 @@ public TextBlock getJavadocBefore(int lineNoBefore) {
         // Lines start at 1 to the callers perspective, so need to take off 2
         int lineNo = lineNoBefore - 2;
 
-        // skip blank lines
-        while (lineNo > 0 && (lineIsBlank(lineNo) || lineIsComment(lineNo))) {
+        while (lineNo > 0 && shouldSkip(lineNo)) {
```

Comment:
uhh, and now we need to do regression on all affected Checks - https://github.com/checkstyle/checkstyle/search?q=getJavadocBefore&type=Code

and provide investigation why such code was added, by git history.

Reply:
I am too lazy to do that :)
In fact, I removed all changes from FileContents class and added one getter. In this case no additional regressions required.